src/solver/external.rs

```diff
@@ -103,18 +105,15 @@ impl FormulaVisitor<Result<SymbolId, SolverError>> for SmtPrinter {
     fn unary(
         &mut self,
         idx: SymbolId,
-        op: BVOperator,
+        _op: BVOperator,
         v: Result<SymbolId, SolverError>,
     ) -> Result<SymbolId, SolverError> {
         let mut o = self.output.lock().expect("no other thread should fail");
 
         writeln!(
             o,
-            "(declare-fun x{} () (_ BitVec 64))\n(assert (= x{} ({} x{})))",
-            idx,
-            idx,
-            to_smt(op),
-            v?
+            "(declare-fun x{} () (_ BitVec 64))\n(assert (= x{} (ite (= x{} (_ bv0 64)) (_ bv1 64) (_ bv0 64))))",
```

Comment:
This seems to specifically implement `BVOperator::Not`, does it make sense to assert `op == BVOperator::Not` in this method to document that we assume it to be the only unary operator.

Reply:
Yes that's a good idea, I implemented that in a new version with an `unreachable!()` in a match, which effectively does exactly the same thing.